lib/menubar/init.lua

```diff
@@ -231,47 +231,74 @@ local function menulist_update(query, scr)
+
+    local PRIO_NONE = 0
+    local PRIO_HIG = 3
+    local PRIO_LOW = 1
+    local PRIO_NORMAL = 2
+
     -- Add the categories
     if menubar.show_categories then
         for _, v in pairs(menubar.menu_gen.all_categories) do
             v.focused = false
             if not current_category and v.use then
                 if string.match(v.name, pattern) then
                     if string.match(v.name, "^" .. pattern) then
-                        table.insert(shownitems, v)
-                    else
-                        table.insert(match_inside, v)
+                        v.count = PRIO_NONE
+                        v.prio = PRIO_NORMAL
+
+                        -- use count from count_table if present
+                        if string.len(pattern) > 0 and count_table[v.name] ~= nil then
+                            v.count = tonumber(count_table[v.name])
+                        end
+
+                        if string.match(v.name, "^" .. pattern)
+                            or string.match(v.cmdline, "^" .. pattern) then
+                            v.prio = PRIO_HIGH
+                        else
+                            v.prio = PRIO_NORMAL
+                        end
+
+                        table.insert (command_list, v)
                     end
                 end
             end
         end
     end
 
-    local count_table = load_count_table()
-    local command_list = {}
-
     -- Add the applications according to their name and cmdline
     for _, v in ipairs(menubar.menu_entries) do
         v.focused = false
         if not current_category or v.category == current_category then
             if string.match(v.name, pattern)
                 or string.match(v.cmdline, pattern) then
-                if string.match(v.name, "^" .. pattern)
-                    or string.match(v.cmdline, "^" .. pattern) then
 
-                    v.count = 0
-                    -- use count from count_table if present
-                    if string.len(pattern) > 0 and count_table[v.name] ~= nil then
-                        v.count = tonumber(count_table[v.name])
-                    end
+                v.count = 0
+                v.prio = PRIO_NONE
+
+                -- use count from count_table if present
+                if string.len(pattern) > 0 and count_table[v.name] ~= nil then
+                    v.count = tonumber(count_table[v.name])
+                end
 
-                    table.insert (command_list, v)
+                if string.match(v.name, "^" .. pattern)
+                    or string.match(v.cmdline, "^" .. pattern) then
+                    v.prio = PRIO_LOW
+                else
+                    v.prio = PRIO_NONE
                 end
+
+                table.insert (command_list, v)
             end
         end
     end
 
     local function compare_counts(a,b)
+        if a.count == b.count then
+            return a.prio > b.prio
+        end
         return a.count > b.count
```

Comment:
This means that first the counts are compared and only then the priorities. What do others think? Should the priority or the count be more important?
I think that it should be the other way around and the priority should win.